bot/cogs/moderation/scheduler.py

```diff
@@ -146,14 +146,18 @@ def mod_log(self) -> ModLog:
                 if expiry:
                     # Schedule the expiration of the infraction.
                     self.schedule_task(ctx.bot.loop, infraction["id"], infraction)
-            except discord.Forbidden:
+            except discord.HTTPException as e:
                 # Accordingly display that applying the infraction failed.
                 confirm_msg = f":x: failed to apply"
                 expiry_msg = ""
                 log_content = ctx.author.mention
                 log_title = "failed to apply"
 
-                log.warning(f"Failed to apply {infr_type} infraction #{id_} to {user}.")
+                log_msg = f"Failed to apply {infr_type} infraction #{id_} to {user}"
+                if isinstance(e, discord.Forbidden):
+                    log.warning(f"{log_msg}: bot lacks permissions.")
+                else:
+                    log.exception(log_msg)
```

Comment:
If it's not a `discord.Forbidden`, there's definitely something unusual going on so it's best to probably log the full traceback also for when we need to investigate.

```suggestion
                    log.exception(log_msg, exc_info=True)
```

Reply:
`.exception()` already sets it to `True` by default.

Reply:
So it does, my mistake.